File: Node/TypeORM_ORM/src/bll/services/orderService.ts
import { OrderRepository } from '../../dal/repositories/orderRepository';
import { Orders } from '../../dal/entities/Orders';

const process = require('process');

export class OrderService {
  private orderRepository = new OrderRepository();

  async getAll() {
    return await this.orderRepository.getAll();
  }

  async getById(id: number) {
    return await this.orderRepository.getById(id);
  }

  async create(entity: Partial<Orders>) {
    return await this.orderRepository.create(entity);
  }

  async update(id: number, entity: Partial<Orders>) {
    await this.orderRepository.update(id, entity);
  }

  async delete(id: number) {
    await this.orderRepository.delete(id);
  }
}

export default OrderService;

File: Go/SQLBoiler_ORM/bll/services/orderService.go
package services

import (
	"SQLBOILER_ORM/dal/entities"
	"SQLBOILER_ORM/dal/repositories"
	"context"
)

type OrderService struct {
	repo *repositories.OrderRepository
}

func NewOrderService(repo *repositories.OrderRepository) *OrderService {
	return &OrderService{repo: repo}
}

func (s *OrderService) GetByID(ctx context.Context, id int) (*entities.Order, error) {
	return s.repo.GetByID(ctx, id)
}

func (s *OrderService) GetAll(ctx context.Context) ([]*entities.Order, error) {
	return s.repo.GetAll(ctx)
}

func (s *OrderService) Create(ctx context.Context, order *entities.Order) (*entities.Order, error) {
	return s.repo.Create(ctx, order)
}

func (s *OrderService) Update(ctx context.Context, order *entities.Order) error {
	return s.repo.Update(ctx, order)
}

func (s *OrderService) Delete(ctx context.Context, id int) error {
	return s.repo.Delete(ctx, id)
}
